### ISSUE TITLE
str2str pass rover coordinates to base station

### ISSUE THREAD

**Issue by Kmaltan** [tomojitakasu/RTKLIB#451]
Hello everyone.
My rover F9P simpleRTK2B
 can get rtk in rtcm3 format, and use base station with VRS and needs to get rover coordinates.
With:
str2str -in ntrip://USER:PASSWD@IP_ADRESS:PORT/MOUNTPOINT -p LAT LOG ALT -n 1 -out serial://ttyS0:57600:8:n:1:off
on a Rasspberry I get RTCM for the Rover.
How could I pass current rover coordinates to base station ?
str2str -in ntrip://USER:PASSWD@IP_ADRESS:PORT/MOUNTPOINT -b -n 1 -out serial://ttyS0:57600:8:n:1:off 
This command don't work.
Positions output from Rover to Raspberry serial:
$GPGGA,121023.04,5034.4267670,N,01025.2441644,E,1,00,1.0,152.796,M,47.204,M,0.0,*48
also i try 
$GNGGA,121023.04,5034.4267670,N,01025.2441644,E,1,00,1.0,152.796,M,47.204,M,0.0,*48
at the serial Port

I try RTKLIB2.4.3 B31 and RTKLIB2.4.3 B28.

**Comment by DavidKelleySCSC** [comment]
Well first you should probably run four plus hours into OPUS (or something similar) to get a position you can believe and then confirm the data is in the same frame of reference you expect the rovers to be using.   You said VRS here but I presume you are trying to create a base station, not to use one. [If you are in fact trying to send NMEA to the Caster to slectd the right base, your Client software wil do that, sending the data you showed below]

As you have deduced, if the base does not provide a message (MT1005/1006) with its precise location , you must setup strstv to do it for you.   Are the three values above filled in with the data when you use the command line?  (in DD.dddd format).  If you can run the GUI, you can also enter it here: 

![setbaseonrtklibstrsvr](https://user-images.githubusercontent.com/5143060/51124755-ee982f00-17d3-11e9-9353-8bf12b302072.png)


Also, I have noticed in our shop we are now seeing a number of F9P devices showing up on the RTK2go open caster and oddly (to me at least) many of these send both RTCM and uBlox message types in the same data stream.  The typical Rover device, expecting RTCM3, cannot deal with that.  I do not know if RTKLIB can deal with it (perhaps others know, if so please reply), I have not had a chance to test that.   [I presume uBlox can, but the information is redundant and wasteful]

**Comment by Kmaltan** [comment]
I not trying to create a base station. I want use a Virtuelle Referenzstation (VRS). Sending a fixed position runs. But str2str not send the rover position to the VRS

**Comment by DavidKelleySCSC** [comment]
Sorry for my confusion.  Is there some reason you cannot set this up with RTKNavi and have it relay the returned $GGA sentences from the F9P to the VRS base? [YOU can then send that stream to Stream server if there is a need.

**Comment by tomojitakasu** [comment]
The -n option of str2str accepts the NMEA GGA cycle not in sec but in ms. (However, less than 1000 ms except for 0 ms or less is recognized as 1000 ms.)

The -b option, which is to relay back messages from the output stream to the input stream, should be followed by the number of the output stream, typically like "-b 1". In this case, you don't need -n option.

Note that the -b option is only supported by 2.4.3 b. It is not yet available in 2.4.2 at this moment.

**Comment by Kmaltan** [comment]
Now the F9P run with Pi Zero and LTE Surfstick for RTK.
Many thanks.

**Comment by morcka** [comment]
Dear Kmaltan,

Could you share the str2str commands, which are working for you?
I would like to use a Raspberry with (USB) F9P for these tasks:
NTRIP VRS data --> F9P
NTRIP VRS data --> log file
F9P GGA --> NTRIP VRS
F9P data --> log file
Do you think that this setup is feasible with the Pi and str2str?

Thank you very much!

**Comment by natsuka81** [comment]
Hi, I've encounter the similar issue and would really appreciate if i get advice from you.

I'm trying to obtain rtk solution with F9P as a moving rover and VRS on rtkrcv but it doesn't work.
How is it possible to send rover position to VRS while using rtkrcv to do rtk?

I thought of a procedure like below and run
$ sudo ./str2str -in ntrip://@USER:PASSWD@IP_ADRESS:PORT/MOUNTPOINT -b 1 -out tcpcli://localhost:9000 -out tcpsvr://:1201

and
write like these in the conf file of rtkrcv 
inpstr1-path       =ttyACM0:115200:8:n:1:off
inpstr2-path       =localhost:1201
outstr1-type       =tcpsvr
outstr1-path       =:9000
outstr1-format     =nmea

but when I ran
$ ./rtkrcv -o rtk_swepos.conf 
I only got single solution and the data from VRS didn't received properly in rtkrcv.

It'd be really appreciated if someone give me any advice.

And I use RTKLIB 2.4.3 b33.

Thank you,


---------------------
Idea
![image](https://user-images.githubusercontent.com/37062686/83818916-cc54c900-a6c8-11ea-8f95-0503fd143cb0.png)
